grafana/grafana
# Explore: Implement logs sample in Explore

**What is this feature?**

**Design:** https://www.figma.com/file/YeTF07V4TUNydMJMD1DYBd/Query-builder-improvements?node-id=3394%3A24058

**User interviews:** https://docs.google.com/document/d/1chSZLp2crXK-aVYE7rtwZkCY-bPT8xnJAOJrblcRlps/edit#heading=h.yt0yfkxblq53

For users who use data source that supports both - logs and metrics (Loki, ElasticSearch), it is beneficial to have simple and quick way how that can see logs that produced visualized metrics. 

For log queries, we already have a concept of seeing metrics results in log volume. Log volume component is toggle-able so you can run and see this queries only in cases where you specifically want to do it. We used the same logic and same UX pattern - implementing toggle-able component that let's you see logs. 

**Why do we need this feature?**

This is beneficial for:
- debugging purposes (looking at specific raw logs)
- writing metrics query, when you want to see what parser or parsed labels are relevant

**Who is this feature for?**

Explore users of data sources, that produce logs and metrics (e.g. Loki, ElasticSearch)

**Which issue(s) does this PR fix?**:

Fixes #

https://github.com/grafana/grafana/issues/52088

https://user-images.githubusercontent.com/30407135/213734430-123861aa-ac9a-4dfc-a505-0ecf41e555d1.mov

**How to test**

This feature currently works only for Loki data source.

1. Run loki e.g. `make devenv sources=loki`
2. Go to Explore
3. Run metric query e,g, `rate({place="moon"}[5m)`
4. You should be able to see both metrics and log sample
5. Toggle off Log samples and change metric query
6. Check in network tab that new query for log samples wasn't run
7. Open Log samples and check that new log sample query was executed and logs are displayed
